src/Illuminate/Database/Schema/Blueprint.php

```diff
@@ -705,14 +705,18 @@ public function boolean($column)
 
     /**
      * Create a new enum column on the table.
+     * If they don't pass an array, assume they want
+     * a dynamic enum.
      *
-     * @param  string  $column
-     * @param  array   $allowed
+     * @param  string      $column
+     * @param  array|null  $allowed
      * @return \Illuminate\Support\Fluent
      */
-    public function enum($column, array $allowed)
+    public function enum($column, array $allowed = null)
```

Comment:
Maybe don't enforce `array`, as we may want to use an object that implement `\ArrayAccess`